frontend/src/ui/Access.tsx

```diff
@@ -0,0 +1,698 @@
+import {
+    useColorScheme,
+    match,
+    ProtoButton,
+    WithTooltip,
+    FloatingHandle,
+    Floating,
+} from "@opencast/appkit";
+import {
+    createContext,
+    forwardRef,
+    useRef,
+    useImperativeHandle,
+    useState,
+    useContext,
+    ReactNode,
+} from "react";
+import { useTranslation } from "react-i18next";
+import { FiX, FiAlertTriangle } from "react-icons/fi";
+import { MultiValue } from "react-select";
+import CreatableSelect from "react-select/creatable";
+import { focusStyle } from ".";
+import { useUser, isRealUser, UserState } from "../User";
+import { COLORS } from "../color";
+import i18n from "../i18n";
+import {
+    DUMMY_GROUPS,
+    DUMMY_USERS,
+    SUBSET_RELATIONS,
+    LARGE_GROUPS,
+} from "../routes/manage/Video/dummyData";
+import { COMMON_ROLES } from "../util/roles";
+import { SelectProps } from "./Input";
+import { searchableSelectStyles, theme } from "./SearchableSelect";
+import { FloatingBaseMenu } from "./FloatingBaseMenu";
+
+export type Acl = {
+    readRoles: string[];
+    writeRoles: string[];
```

Comment:
Seeing how many `[...old, role]` and `old.filter(v => v != role)` are used below, I think this should actually be:

```suggestion
    readRoles: Set<string>;
    writeRoles: Set<string>;
```

That should make the code a lot easier to read as you can just use `add` and `delete`. Also, there might be a few O(n²) algorithms below, which could actually be a problem with huge user lists. I haven't found one specifically, but my spider senses were tickling. 

"Consumers" of this type should be able to easily convert it to arrays again via `[...readRoles]`.

Only problem: just calling `add` will not work with reacts state management. We still have to call `setSelections`, but I think this should always work: `setSelections(old => { let out = new Set(old); out.add(...); return out })`. One could write this once in some kind of helper function.